scripts/qa_crowbarsetup.sh

```diff
@@ -2303,6 +2355,19 @@ function custom_configuration
             if iscloudver 7plus ; then
                 proposal_set_value nova default "['attributes']['nova']['metadata']['vendordata']['json']" "'{\"custom-key\": \"custom-value\"}'"
             fi
+
+            if iscloudver 7plus && [[ $want_ironic = 1 ]] ; then
+                local nova_controller
+                if [[ $hacloud = 1 ]] ; then
+                    # Ironic can't be deployed as an HA API service, but we can use one of the nova controllers in the cluster
+                    local pacemaker_proposal_type=`python -c "import json; f=open('$pfile'); j=json.load(f); print j['deployment']['nova']['elements']['nova-controller'][0].split(':')[1]"`
+                    local pacemaker_pfile=`get_proposal_filename "pacemaker" "$pacemaker_proposal_type"`
+                    nova_controller=`python -c "import json; f=open('$pacemaker_pfile'); j=json.load(f); print j['deployment']['pacemaker']['elements']['pacemaker-cluster-member'][0]"`
+                else
+                    nova_controller=`python -c "import json; f=open('$pfile'); j=json.load(f); print j['deployment']['nova']['elements']['nova-controller'][0]"`
```

Comment:
this looks pretty similar to the existing get_novacontroller which uses rubyjsonparse
and resolve_element_to_hostname might also help in the HA case